tests/integration/test_dashNgl.py

```diff
@@ -0,0 +1,215 @@
+import glob
+import time
+
+import dash
+from dash.dependencies import Input, Output, State
+import dash_html_components as html
+import dash_bio
+
+from common_features import simple_app_layout
+
+_COMPONENT_ID = "test-ngl"
+data_path = "tests/dashbio_demos/dash-ngl/data/"
+
+color_list = ["red", "blue"]
+
+# PDB examples
+dropdown_options = [
+    "1PNK",
+    "6CHG",
+    "3K8P",
+]
+
+# Placeholder which is loaded if no molecule is selected
+data_dict = {
+    "selectedValue": "placeholder",
+    "chain": "ALL",
+    "color": "#e41a1c",
+    "filename": "placeholder",
+    "ext": "",
+    "config": {"type": "text/plain", "input": ""},
+}
+
+
+# Helper function to load the data
+def get_data(selection, pdb_id, color):
+
+    chain = "ALL"
+
+    # Check if only one chain should be shown
+    if "." in pdb_id:
+        pdb_id, chain = pdb_id.split(".")
+
+    if pdb_id not in dropdown_options:
+        return data_dict
+    else:
+        fname = [f for f in glob.glob(data_path + pdb_id + ".*")][0]
+
+        with open(fname, "r") as f:
+            contents = f.read()
+
+        return {
+            "selectedValue": selection,
+            "chain": chain,
+            "color": color,
+            "filename": fname.split("/")[-1],
+            "ext": fname.split(".")[-1],
+            "config": {"type": "text/plain", "input": contents},
+        }
+
+
+viewer = html.Div(
+    [dash_bio.DashNgl(
+        id=_COMPONENT_ID,
+        data=data_dict)],
+    style={
+        "display": "inline-block",
+        "width": "calc(100% - 500px)",
+        "float": "left",
+        "marginTop": "50px",
+        "marginRight": "50px",
+    },
+)
+
+
+# Based on simple_app_callback
+# simple_app_callback does not work because:
+# 1.DashNgl does not accept the output as a string
+# 2.Two submissions are needed to pass shouldComponentUpdate in DashNgl
```

Comment:
Could you elaborate on this as well?

Reply:
When the app loads an empty dictionary is  loaded with the selectedValue:placeholder
see the console.log screenshots of my app "That is a weird bug if you console.log (this.props.data) during the start of the app its properties are changing. See in the last line in the picture below"

So the thing which has to be passed is:
```
const oldSelection = this.props.data[0].selectedValue;
const newSelection = nextProps.data[0].selectedValue;
if (oldSelection !== newSelection) {
            return true;
        }
```
It is not possible to initialise the container correctly w/o any
dummy value at least I did not manage it

Reply:
I think it might be because you accept both a dictionary/object and a list/array for the `data` prop: https://github.com/IvoLeist/dash-bio/blob/e47ceba4057d00138bec2bf7febb678c2d539bb2/src/lib/components/DashNgl.react.js#L213

You could define a default value as an empty dict or an empty list here: https://github.com/IvoLeist/dash-bio/blob/e47ceba4057d00138bec2bf7febb678c2d539bb2/src/lib/components/DashNgl.react.js#L180

Reply:
@shammamah adding a default dictionary was a really good idea it resolved that issue :)
& this: 

> When the app loads an empty dictionary is loaded with the selectedValue:placeholder
> see the console.log screenshots of my app "That is a weird bug if you console.log (this.props.data) during the start of the app its properties are changing. See in the last line in the picture below"

as well :) !